src/lib/conda.ml

```diff
@@ -0,0 +1,123 @@
+(*
+  Conda is a Python environment and package manager:
+  http://conda.pydata.org/docs/
+
+  We use it to ensure a consistent Python environment for tools that depend
+  on Python.
+*)
+
+open Common
+open Run_environment
+
+let rm_path = Workflow_utilities.Remove.path_on_host
+
+let dir ~meta_playground = meta_playground // "conda_dir"
+let commands ~meta_playground com = dir ~meta_playground // "bin" // com
+let bin = commands "conda"
+let activate = commands "activate"
+let deactivate = commands "deactivate"
+
+(* give a conda command. *)
+let com ~meta_playground fmt =
+  Printf.sprintf ("%s " ^^ fmt) (bin ~meta_playground)
+
+(* A workflow to ensure that conda is installed. *)
+let installed ~host ~meta_playground =
+  let open KEDSL in
+  let url =
+    "https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh"
+  in
+  let conda_exec  = single_file ~host (bin ~meta_playground) in
+  let install_dir = dir ~meta_playground in
+  workflow_node conda_exec
+    ~name:"Install conda"
+    ~make:(daemonize ~host
+             Program.(exec ["cd"; meta_playground]
+                      && Workflow_utilities.Download.wget_program url
+                                            (* -b : batch be silent -p prefix *)
+                      && shf "bash Miniconda3-latest-Linux-x86_64.sh -b -p %s"
+                                install_dir))
+
+let config = "biokepi_conda_env"
+let env_name = "biokepi"
+
+(* Ensure that a file exists describing the default linux conda enviroment.
+   TODO: figure out a better solution to distribute this configuration.  *)
+let cfg_exists ~host ~meta_playground =
+  let open KEDSL in
+  let file = meta_playground // config in
+  let make = daemonize ~host
+      Program.(shf "echo '%s' >> %s"
+{python|# This file may be used to create an environment using:
```

Comment:
`"%s" (Filename.quote {python|....` would be more future-proof than `"'%s'" ...` because it's easy to add single-quotes.

Also why `{python| |python}`, is this python syntax?